chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -15,9 +15,42 @@ import chisel3.internal.sourceinfo._
   * of) other Data objects.
   */
 sealed abstract class Aggregate extends Data {
+  protected def childLitArgs = getElements.map(_.litArg).foldLeft[Option[Seq[LitArg]]](Some(Seq[LitArg]())) {
+    case (Some(elems), Some(lit)) => Some(elems :+ lit)
+    case _                        => None
+  }
+  override def litArg = {
+    // TODO[grebe] make a lit for Aggregates
+    childLitArgs.map { x => ULit(0, 1.W) }
+  }
+
+  private var checkedBinding = false
+  private def checkBinding = {
+    // check if this aggregate is a literal, and bind it if it hasn't been bound yet
+    if (!checkedBinding && getElements.length > 0 && childLitArgs.isDefined) {
+      this.bind(LitBinding()) //, SpecifiedDirection.Unspecified)
+    }
+    checkedBinding = true
+  }
+
+  protected[core] override def hasBinding = {
+    checkBinding
```

Comment:
This looks kind of hacky, kind of like a game of whackamole to insert checks into all the right places, it would make more sense for higher-level code to apply the binding as it is being set to a literal.

Reply:
The problem is that bind() is `private[chisel3]` which makes it impossible to do that in user-land right now.